src/components/rangeslider/draw.js

```diff
@@ -82,7 +82,7 @@ module.exports = function(gd) {
         // compute new slider range using axis autorange if necessary
         // copy back range to input range slider container to skip
         // this step in subsequent draw calls
-        if(!opts.range) {
+        if(axisOpts._needsExpand && axisOpts._min.length && axisOpts._max.length) {
```

Comment:
As the computed range slider range (i.e. `opts.range` here) is copied to the user layer container, `Axes.getAutoRange` would not get called on data array updates (unless one would `delete gd.layout.xaxis.rangeslider.range`).

Note that `xaxis._min` and `xaxis._max` are only defined when an update call passed through the `calc` step as  `setConvert` clears them [here](https://github.com/plotly/plotly.js/blob/e7f69c986ba11e6672b0ff55bc76389c934d64cb/src/plots/cartesian/set_convert.js#L405-L409) in supplyDefaults.